### PR TITLE
refactor(functional-tests): distribute functional tests across workers evenly (part 2)

### PR DESCRIPTION
**Pull request by Trinaa**
## Because

- we want to execute the functional tests in CI as fast as possible

## This pull request
- matches parallelism to the number of [cores for a resource_class](https://circleci.com/product/features/resource-classes/) (4 for large and 8 for xlarge)
- splits and refactors the `misc/keyStretchingV2.spec` module into:
  - `key-stretching-v2/totp.spec.ts`
  - `key-stretching-v2/signinBlocked.spec.ts`
- splits the `oauth/oauthPermissions.spec.ts` module into:
  - `oauth/oauthPermissionsSignin.spec.ts`
  - `oauth/oauthPermissionsSignup.spec.ts`
- splits part of the `oauth/signin.spec.ts` module into `oauth/signinBlocked.spec.ts`
- splits part of the `react-conversion/oauthResetPassword.spec.ts` module into:
  - `react-conversion/oauthResetPasswordRecoveryKey.spec.ts`
  - `react-conversion/oauthResetPasswordScopeKey.spec.ts`
  - `react-conversion/oauthResetPasswordSyncMobile.spec.ts`
- splits part of the `react-conversion/signup.spec.ts` module into `react-conversion/oauthSignup.spec.ts`
- splits part of the `settings/changeEmail.spec.ts` module into `settings/changeEmailBlocked.spec.ts`
- splits part of the `settings/changePassword.spec.ts` module into `settings/changePasswordValidation.spec.ts`
- splits part of the `subscription-tests/coupons-ui-tests/coupon.spec.ts` module into:
  - `subscription-tests/coupons-ui-tests/couponExpired.spec.ts` 
  - `subscription-tests/coupons-ui-tests/couponForeverDiscount.spec.ts`
  - `subscription-tests/coupons-ui-tests/couponInvalid.spec.ts`
  - `subscription-tests/coupons-ui-tests/couponOneTimeDiscount.spec.ts`
- splits part of the `subscription-tests/coupons-ui-tests/resubscription.spec.ts` module into `subscription-tests/coupons-ui-tests/paymentMode.spec.ts`
- splits part of the `syncV3/fxDesktopHandshake.spec.ts` module into `syncV3/fxDesktopHandshakeNonSync.spec.ts`
- splits part of the `syncV3/fxDesktopV3ForceAuth.spec.ts` module into `syncV3/fxDesktopV3ForceAuthUnregistered.spec.ts`
- splits part of the `syncV3/signIn.spec.ts` module into `syncV3/oauthSignIn.spec.ts`
- fixes `confirmResetPasswordHeading` error in `react-conversion/resetPassword.spec.ts`

## Issue that this pull request solves

Closes: # FXA-9699

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

Before (Nightly May 23 2024) ~ 6m 53s 
image

Best After ~ 6m 14s (no retries)
image

Notes 
- The fixme rate jumped from 2-14% on this May 23rd, which makes understanding the impact of this PR hard to measure
- The CI is configured to use historical timing data to optimize the parallel workers, so we may see better distribution over time

## Other information (Optional)

Any other information that is important to this pull request.

Files with code changes include the following, all other changes are straight-forward splits (no code changes):
- .circleci/config.yml
- packages/functional-tests/pages/signinTotpCode.ts
- packages/functional-tests/tests/key-stretching-v2/signinBlocked.spec.ts
- packages/functional-tests/tests/key-stretching-v2/totp.spec.ts
- packages/functional-tests/tests/react-conversion/resetPassword.spec.ts

### CODE REVIEW

File: packages/functional-tests/tests/key-stretching-v2/signinBlocked.spec.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+/**
+ * These tests represent various permutations between interacting with V1 and V2
+ * key stretched passwords. We need to ensure that operations are interchangeable!
+ */
+test.describe('severity-2 #smoke', () => {
+  type Version = { version: 1 | 2; query: string };
+  type TestCase = { signup: Version; signin: Version };
+  const v1: Version = { version: 1, query: '' };
+  const v2: Version = { version: 2, query: 'stretch=2' };
```
Comment (pdehaan) [inline packages/functional-tests/tests/key-stretching-v2/signinBlocked.spec.ts:21]: For (Part 3), i wonder if we should export these so we can share them between files:
```sh
git grep -Pn "type (TestCase|Version) = " packages | cat

packages/functional-tests/tests/key-stretching-v2/changePassword.spec.ts:31:  type Version = { version: 1 | 2; query: string };
packages/functional-tests/tests/key-stretching-v2/changePassword.spec.ts:32:  type TestCase = { signup: Version; change: Version };
packages/functional-tests/tests/key-stretching-v2/recoveryKey.spec.ts:35:  type Version = { version: 1 | 2; query: string };
packages/functional-tests/tests/key-stretching-v2/recoveryKey.spec.ts:36:  type TestCase = { signup: Version; reset: Version; signin: Version };
packages/functional-tests/tests/key-stretching-v2/resetPassword.spec.ts:34:  type Version = { version: 1 | 2; query: string };
packages/functional-tests/tests/key-stretching-v2/resetPassword.spec.ts:35:  type TestCase = { signup: Version; reset: Version; signin: Version };
packages/functional-tests/tests/key-stretching-v2/signinBlocked.spec.ts:18:  type Version = { version: 1 | 2; query: string };
packages/functional-tests/tests/key-stretching-v2/signinBlocked.spec.ts:19:  type TestCase = { signup: Version; signin: Version };
packages/functional-tests/tests/key-stretching-v2/totp.spec.ts:32:  type Version = { version: 1 | 2; query: string };
packages/functional-tests/tests/key-stretching-v2/totp.spec.ts:33:  type TestCase = { signup: Version; signin: Version };
```

&hellip; aaactually, it looks like `TestCase` can be different in some files, so only `Version` could be exported. Wonder if it's even worth it.

**Review by pdehaan — COMMENTED** [review]
Excited to see how much more speed this should shave off. 🤞⏱️

File: packages/functional-tests/tests/key-stretching-v2/signinBlocked.spec.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  const TestCases: TestCase[] = [
+    { signup: v1, signin: v1 },
+    { signup: v1, signin: v2 },
+    { signup: v2, signin: v1 },
+    { signup: v2, signin: v2 },
+  ];
+
+  for (const { signup, signin } of TestCases) {
+    test(`signs up as v${signup.version}, rate limited, unblocked, signs in as v${signin.version} for backbone`, async ({
```
[the remark this answers is not in the document]
Comment (Trinaa) [reply packages/functional-tests/tests/key-stretching-v2/signinBlocked.spec.ts:30]: I think signup is rolled out for react, but signin isn't fully rolled out, so the part that's backbone is signin.

File: packages/functional-tests/tests/key-stretching-v2/signinBlocked.spec.ts (hunk 1 above)
Comment (Trinaa) [reply packages/functional-tests/tests/key-stretching-v2/signinBlocked.spec.ts:21]: I'm inclined to leave them honestly. It's verbose, but I think easier to understand if we define them in the test module.

**Review by vpomerleau — APPROVED** [review]
LGTM to split up these tests into smaller test files 👍

**Review by pdehaan — APPROVED** [review]

**Review by ashrivastava-qa — APPROVED** [review]
LGTM !

**Pull request merged**